src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,105 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    static RegistrationService registrationService;
+    static User normalUser1;
+    static User normalUser2;
+    static User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        normalUser1 = new User();
+        normalUser1.setLogin("Vasya");
+        normalUser1.setAge(19);
+        normalUser1.setPassword("abracadabra");
+        normalUser2 = new User();
+        normalUser2.setLogin("Kate");
+        normalUser2.setAge(33);
+        normalUser2.setPassword("MAGA2024");
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        testUser = new User();
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        testUser.setLogin("Fedir");
+        testUser.setAge(null);
+        testUser.setPassword("OctoPussy");
+        assertThrows(RuntimeException.class, () ->
+            registrationService.register(testUser));
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        testUser = null;
+        assertThrows(RuntimeException.class, () ->
+            registrationService.register(testUser));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        testUser.setLogin("Anna");
+        testUser.setAge(19);
+        testUser.setPassword("qwer");
+        assertThrows(RuntimeException.class, () ->
+            registrationService.register(testUser));
+    }
+
+    @Test
+    void register_underage_notOk() {
```

Comment:
```suggestion
    void register_underAge_notOk() {
```

Reply:
"underage - (of a person) too young to engage legally in a particular activity, especially drinking alcohol or having sex.")

Reply:
sorry) my bad